File: rust/src/models/invoice_line_item.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::sub_line_item::SubLineItem;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct InvoiceLineItem {
    pub amount_total: i32,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    pub end_date: String,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub quantity: Option<String>,

    pub start_date: String,

    pub sub_line_items: Vec<SubLineItem>,

    pub tax_rate: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub unit_price: Option<String>,
}

impl InvoiceLineItem {
    pub fn new(
        amount_total: i32,
        end_date: String,
        name: String,
        start_date: String,
        sub_line_items: Vec<SubLineItem>,
        tax_rate: String,
    ) -> Self {
        Self {
            amount_total,
            description: None,
            end_date,
            name,
            quantity: None,
            start_date,
            sub_line_items,
            tax_rate,
            unit_price: None,
        }
    }
}

File: typescript/src/models/invoiceLineItem.ts
// this file is @generated
import { type SubLineItem, SubLineItemSerializer } from "./subLineItem";

export interface InvoiceLineItem {
  amountTotal: number;

  description?: string | null;

  endDate: string;

  name: string;

  quantity?: string;

  startDate: string;

  subLineItems: SubLineItem[];

  taxRate: string;

  unitPrice?: string;
}

export const InvoiceLineItemSerializer = {
  _fromJsonObject(object: any): InvoiceLineItem {
    return {
      amountTotal: object["amount_total"],
      description: object["description"],
      endDate: object["end_date"],
      name: object["name"],
      quantity: object["quantity"],
      startDate: object["start_date"],
      subLineItems: object["sub_line_items"].map((item: any) =>
        SubLineItemSerializer._fromJsonObject(item)
      ),
      taxRate: object["tax_rate"],
      unitPrice: object["unit_price"],
    };
  },

  _toJsonObject(self: InvoiceLineItem): any {
    return {
      amount_total: self.amountTotal,
      description: self.description,
      end_date: self.endDate,
      name: self.name,
      quantity: self.quantity,
      start_date: self.startDate,
      sub_line_items: self.subLineItems.map((item: any) =>
        SubLineItemSerializer._toJsonObject(item)
      ),
      tax_rate: self.taxRate,
      unit_price: self.unitPrice,
    };
  },
};
